packages/esm-patient-chart-app/src/clinical-views/hooks/useLastEncounter.ts

```diff
@@ -0,0 +1,27 @@
+import { openmrsFetch } from '@openmrs/esm-framework';
+import { type Encounter } from '../types';
+import useSWRImmutable from 'swr/immutable';
+
+const encounterRepresentation =
+  'custom:(uuid,encounterDatetime,encounterType,location:(uuid,name),' +
+  'patient:(uuid,display,age,identifiers,person),encounterProviders:(uuid,provider:(uuid,name)),' +
+  'obs:(uuid,obsDatetime,voided,groupMembers,concept:(uuid,display,name:(uuid,name)),value:(uuid,name:(uuid,name,display),' +
+  'names:(uuid,conceptNameType,name,display))),form:(uuid,name))';
+
+export function useLastEncounter(patientUuid: string, encounterType: string) {
+  const query = `encounterType=${encounterType}&patient=${patientUuid}&limit=1&order=desc&startIndex=0`;
```

Comment:
I'm confused as to why this query includes fixed parts, but then we append a semidynamic value (the representation) to the end. This feels like it could be better organized.

Reply:
Would this suffice ?
```
const query = `encounterType=${encounterType}&patient=${patientUuid}`;
  const endpointUrl =
    patientUuid && encounterType
      ? `/ws/rest/v1/encounter?${query}&v=${encounterRepresentation}&limit=1&order=desc&startIndex=0`
      : null;
```

Reply:
Yeah, that's better.